src/util/rules.js

```diff
@@ -1,60 +1,65 @@
-/* Object mapping token types to pretty names and colors for display */
-export const rules = {
-  // Non-terminal rules
-  'and_expr': { prettyName: 'And', color: '#F0E68C' },
-  'argument': { prettyName: 'Arguments', color: '#EAF7AB' },
-  'arith_expr': { prettyName: 'Arithmetic Expression', color: '#FFA500' },
-  'augassign':   { prettyName: 'Augmented Assignment', color: '#00ffff' },
-  'break_stmt': { prettyName: 'Break Statement', color: '#00ffa0' },
-  'classdef': { prettyName: 'Class Definitions', color: '#03C03C' },
-  'comp_op':  { prettyName: 'Comparison Operator', color: '#DAA520' },
-  'dictorsetmaker': { prettyName: 'Dictionary', color: '#00FF7F' },
-  'except_clause':  { prettyName: 'Except Clauses', color: '#779ECB' },
-  'expr':     { prettyName: 'Expressions', color: '#F0ABF7' },
-  'for_stmt': { prettyName: 'For Loops', color: '#F7ABAB' },
-  'funcdef':  { prettyName: 'Function Definitions', color: '#AEC6CF' },
-  'if_stmt':  { prettyName: 'If Statements', color: '#FFFF00' },
-  'import_name': { prettyName: 'Import Statement', color: '#FF6666' },
-  'integer':  { prettyName: 'Integers', color: '#ABF7C6' },
-  'number':   { prettyName: 'Numbers', color: '#DEA5A4' },
-  'parameters':  { prettyName: 'Parameters', color: '#FFB347' },
-  'pass_stmt':   { prettyName: 'Pass Statements', color: '#FDFD96' },
-  'return_stmt': { prettyName: 'Return Statement', color: '#EECCFF' },
-  'str':      { prettyName: 'Strings', color: '#CAABF7' },
-  'try_stmt': { prettyName: 'Try Statements', color: '#CCCCFF' },
-  'while_stmt':  { prettyName: 'While Loops', color: '#F0E68C' },
-  // Terminal rules
-  '.TRUE': { prettyName: 'Boolean (True)', color: '#FF9933' },
-  '.FALSE': { prettyName: 'Boolean (False)', color: '#E67300' },
+import axios from 'axios'
+import { setRules, setIgnoredRules } from './codemirror-utils';
+
+const API_URL = process.env.REACT_APP_API_URL
+
+const setAllRules = (allRules) => {
+  const {rules, ignoredRules} = allRules;
+  setRules(rules);
+  setIgnoredRules(ignoredRules);
+}
+let store;
+let language = '';
+let mappingCache = {};
+
+// Load csv and parse to rules and ignoredRules objects
+const loadRules = async (language) => {
+  const allRules = await axios.get(`${API_URL}/mappings/${language}`)
+    .then((res) => {
+      const csv = res.data;
+      const rows = csv.split('\n')
+        .slice(1, -1)
+        .map((row) => row.split(','));
+      const reducer = (map, row) => {
+        map[row[0]] = {
+          prettyName: row[2],
+          color: row[3],
+        };
+        return map;
+      }
+      const rules = rows.filter((row) => row[1] === '1')
+        .reduce(reducer, {});
+      const ignoredRules = rows.filter((row) => row[1] === '0')
+        .map(row => row[0]);
+      return {rules, ignoredRules}
+    });
```

Comment:
The point is that that is less code that we have to maintain. Did you write tests to ensure you properly parse CSVs? What if there's a major change to file structure? Using an external library abstracts these things so that we don't have to think about these things.